File: backoffice/tests/shared/StringValueObjectTest.py
"""
 *
 * Libraries 
 *
"""

from unittest          import TestCase
from src.shared.domain import StringValueObject

"""
 *
 * Class
 *
"""

class StringValueObjectTest( TestCase ):

    """
     *
     * Methods 
     *
    """

    def testIsNotAnEqualObject( self ) -> None:
        # Variables
        valueObject : StringValueObject
        response    : bool
        # Code
        valueObject = StringValueObject( 'hello world' )
        response    = valueObject.equals( StringValueObject( 'hello world!' ) )
        self.assertFalse( response )
    
    def testIsAnEqualObject( self ) -> None:
        # Variables
        valueObject : StringValueObject
        response    : bool
        # Code
        valueObject = StringValueObject( 'hello world!' )
        response    = valueObject.equals( StringValueObject( 'hello world!' ) )
        self.assertTrue( response )
    
    def testIsNotAnEqualObjectBecauseItIsNotAnComparableObject( self ) -> None:
        # Variables
        valueObject : StringValueObject
        response    : bool
        # Code
        valueObject = StringValueObject( 'hello world!' )
        response    = valueObject.equals( 'hello world!' )
        self.assertFalse( response )
    
    def testIsNotAnEqualObjectBecauseItIsNotAnComparableObject2( self ) -> None:
        # Variables
        valueObject : StringValueObject
        response    : bool
        # Code
        valueObject = StringValueObject( 'hello world!' )
        response    = valueObject.equals( None )
        self.assertFalse( response )
    
    def testIsAnEmptyObject( self ) -> None:
        # Variables
        valueObject : StringValueObject
        response    : bool
        # Code
        valueObject = StringValueObject( '' )
        response    = valueObject.isEmpty()
        self.assertTrue( response )
    
    def testIsNotAnEmptyObject( self ) -> None:
        # Variables
        valueObject : StringValueObject
        response    : bool
        # Code
        valueObject = StringValueObject( 'hello world!' )
        response    = valueObject.isEmpty()
        self.assertFalse( response )
    
    def testIsAnObjectLongerThan( self ) -> None:
        # Variables
        valueObject : StringValueObject
        response    : bool
        # Code
        valueObject = StringValueObject( 'hello world!' )
        response    = valueObject.isLongerThan( StringValueObject( 'hello world' ) )
        self.assertTrue( response )
    
    def testIsNotAnObjectLongerThan( self ) -> None:
        # Variables
        valueObject : StringValueObject
        response    : bool
        # Code
        valueObject = StringValueObject( 'hello world' )
        response    = valueObject.isLongerThan( StringValueObject( 'hello world!' ) )
        self.assertFalse( response )
    
    def testIsNotAnObjectLongerThan2( self ) -> None:
        # Variables
        valueObject : StringValueObject
        response    : bool
        # Code
        valueObject = StringValueObject( 'hello world!' )
        response    = valueObject.isLongerThan( StringValueObject( 'hello world!' ) )
        self.assertFalse( response )
    
    def testIsAnObjectLongerThanOrEqual( self ) -> None:
        # Variables
        valueObject : StringValueObject
        response    : bool
        # Code
        valueObject = StringValueObject( 'hello world!' )
        response    = valueObject.isLongerThanOrEqual( StringValueObject( 'hello world' ) )
        self.assertTrue( response )
    
    def testIsAnObjectLongerThanOrEqual2( self ) -> None:
        # Variables
        valueObject : StringValueObject
        response    : bool
        # Code
        valueObject = StringValueObject( 'hello world' )
        response    = valueObject.isLongerThanOrEqual( StringValueObject( 'hello world' ) )
        self.assertTrue( response )
    
    def testIsNotAnObjectLongerThanOrEqual( self ) -> None:
        # Variables
        valueObject : StringValueObject
        response    : bool
        # Code
        valueObject = StringValueObject( 'hello world' )
        response    = valueObject.isLongerThanOrEqual( StringValueObject( 'hello world!' ) )
        self.assertFalse( response )

File: preparation/tests/shared/StringValueObjectTest.cs
/* 
 * 
 * Libraries
 *
*/

using NUnit.Framework;
using preparation.src.shared.domain;

/* 
 * 
 * Class
 *
*/

namespace preparation.tests.shared {

    [TestFixture]
    public class StringValueObjectTest {

        /* 
         * 
         * Methods
         *
        */

        [Test]
        public void testIsNotAnEqualObject() {
            // Variables
            StringValueObject valueObject;
            bool              response;
            // Code
            valueObject = new StringValueObject( "hello world" );
            response    = valueObject.equals( new StringValueObject( "hello world!" ) );
            Assert.IsFalse( response );
        }

        [Test]
        public void testIsAnEqualObject() {
            // Variables
            StringValueObject valueObject;
            bool              response;
            // Code
            valueObject = new StringValueObject( "hello world!" );
            response    = valueObject.equals( new StringValueObject( "hello world!" ) );
            Assert.IsTrue( response );
        }

        [Test]
        public void testIsAnEmptyObject() {
            // Variables
            StringValueObject valueObject;
            bool              response;
            // Code
            valueObject = new StringValueObject( "" );
            response    = valueObject.isEmpty();
            Assert.IsTrue( response );
        }

        [Test]
        public void testIsNotAnEmptyObject() {
            // Variables
            StringValueObject valueObject;
            bool              response;
            // Code
            valueObject = new StringValueObject( "hello world!" );
            response    = valueObject.isEmpty();
            Assert.IsFalse( response );
        }

        [Test]
        public void testIsAnObjectLongerThan() {
            // Variables
            StringValueObject valueObject;
            bool              response;
            // Code
            valueObject = new StringValueObject( "hello world!" );
            response    = valueObject.isLongerThan( new StringValueObject( "hello world" ) );
            Assert.IsTrue( response );
        }

        [Test]
        public void testIsNotAnObjectLongerThan2() {
            // Variables
            StringValueObject valueObject;
            bool              response;
            // Code
            valueObject = new StringValueObject( "hello world!" );
            response    = valueObject.isLongerThan( new StringValueObject( "hello world!" ) );
            Assert.IsFalse( response );
        }

        [Test]
        public void testIsAnObjectLongerThanOrEqual() {
            // Variables
            StringValueObject valueObject;
            bool              response;
            // Code
            valueObject = new StringValueObject( "hello world!" );
            response    = valueObject.isLongerThan( new StringValueObject( "hello world" ) );
            Assert.IsTrue( response );
        }

        [Test]
        public void testIsAnObjectLongerThanOrEqual2() {
            // Variables
            StringValueObject valueObject;
            bool              response;
            // Code
            valueObject = new StringValueObject( "hello world" );
            response    = valueObject.isLongerThanOrEqual( new StringValueObject( "hello world" ) );
            Assert.IsTrue( response );
        }

        [Test]
        public void testIsNotAnObjectLongerThanOrEqual() {
            // Variables
            StringValueObject valueObject;
            bool              response;
            // Code
            valueObject = new StringValueObject( "hello world" );
            response    = valueObject.isLongerThanOrEqual( new StringValueObject( "hello world!" ) );
            Assert.IsFalse( response );
        }

    }

}